src/sys/linux/pipe.rs

```diff
@@ -2,4 +2,63 @@
 #![allow(unused)]
 
 pub enum read2 { }
-pub enum AnonPipe { }
+use sync::atomic::{AtomicBool, Ordering};
+use cmp;
+use io;
+use linux;
+use mem;
+use ptr;
+use sys::{cvt, errno};
+use sys::fd::FileDesc;
+
+////////////////////////////////////////////////////////////////////////////////
+// Anonymous pipes
+////////////////////////////////////////////////////////////////////////////////
+
+pub struct AnonPipe(FileDesc);
+
+pub fn anon_pipe() -> io::Result<(AnonPipe, AnonPipe)> {
+    let mut fds = [0; 2];
+
+    // Unfortunately the only known way right now to create atomically set the
```

Comment:
I saw similar comment in stdlib/sys/unix/net.rs. weak!() macro from sys/unix/weak.rs is used there. Can we use those macro here?